documentation/dataplane/dash-flow/DASH_FLOW_SAI.md

```diff
@@ -0,0 +1,615 @@
+
+
+# README
+
+DASH Flow SAI abstracts the flow state within DASH. In the quest for achieving cloud services, the control plane can generate a flow state table, performing operations such as adding, deleting, modifying, and querying. This specification guarantees alignment between the flow state table and flow states across different cloud providers and vendors. We propose an API to provide a vendor-neutral method for managing flow states, including uniform control of Programmable switches, SmartNICs, and Smart Switches.
+
+DASH Flow SAI accommodates multiple flow tables and diverse flow entry operations. Primarily, it negates differences between vendors' implementations, offering universal interfaces. However, it acknowledges performance variations between DASH and SAI deployed on smart devices. For example, it supplies APIs to handle single and batch flows, with entry count as a limiting factor.
+
+With DASH Flow SAI, it's possible to achieve varied cloud services requiring flow states. For instance, a cloud load balancer can add a flow decision entry and retrieve the flow entry via the data plane. It also lays the groundwork for DASH to provide basic services, such as high availability.
+
+
+
```

Comment:
to help with the format problem, we can install a few vscode extensions to make things easy:
- Markdown All in One
- markdownlint

Then we can auto format the doc.